tests/unit/lms/views/admin/application_instance_test.py

```diff
@@ -358,6 +358,73 @@ def test_update_instance_save_ai_fields_keeps_existing(
 
         assert getattr(application_instance, setting) == existing_value
 
+    def test_update_instance_deployment_id(
+        self, pyramid_request, application_instance_lti_13
+    ):
+        pyramid_request.matchdict["consumer_key"] = sentinel.consumer_key
+        pyramid_request.params["deployment_id"] = "NEW_DEPLOYMENT_ID"
+
+        AdminApplicationInstanceViews(pyramid_request).update_instance()
+
+        assert pyramid_request.session.peek_flash("errors")
+        application_instance_lti_13.deployment_id = "NEW_DEPLOYMENT_ID"
+
+    def test_update_instance_deployment_id_same_value(
+        self, pyramid_request, application_instance_lti_13, application_instance_service
+    ):
+        existing_deployment_id = application_instance_lti_13.deployment_id
+        application_instance_service.get_by_deployment_id.return_value = (
+            application_instance_lti_13
+        )
+        pyramid_request.matchdict["consumer_key"] = sentinel.consumer_key
+        pyramid_request.params["deployment_id"] = existing_deployment_id
+
+        AdminApplicationInstanceViews(pyramid_request).update_instance()
+
+        assert not pyramid_request.session.peek_flash("errors")
+        application_instance_lti_13.deployment_id = existing_deployment_id
+
+    def test_update_instance_deployment_id_no_duplicate(
+        self, pyramid_request, application_instance_service, application_instance_lti_13
+    ):
+        application_instance_service.get_by_id.return_value = (
+            application_instance_lti_13
+        )
+        application_instance_service.get_by_deployment_id.side_effect = (
+            ApplicationInstanceNotFound
+        )
+        pyramid_request.params["deployment_id"] = "NEW_DEPLOYMENT_ID"
+
+        AdminApplicationInstanceViews(pyramid_request).update_instance()
+
+        assert not pyramid_request.session.peek_flash("errors")
+        application_instance_lti_13.deployment_id = "NEW_DEPLOYMENT_ID"
+
+    def test_update_instance_duplicated_deployment_id(
+        self,
+        db_session,
+        pyramid_request,
+        application_instance_lti_13,
+        application_instance_service,
+    ):
+        existing_ai = factories.ApplicationInstance(
+            lti_registration_id=application_instance_lti_13.lti_registration_id,
+            deployment_id="NEW_DEPLOYMENT_ID",
+        )
```

Comment:
Above we use `existing_ai.deployment_id`. Setting it to `NEW_DEPLOYMENT_ID` is a bit confusing, as it's the old one?